### ISSUE TITLE
Build error for cmdline.h

### ISSUE THREAD

**Issue by xkszltl** [halide/Halide#4265]
This issue was probably introduced by this commit: https://github.com/halide/Halide/commit/c92a1f6f693fc71ce73580454aa308a36c81aca3
@steven-johnson Any idea?

```
In file included from ../apps/autoscheduler/retrain_cost_model.cpp:12:
../apps/autoscheduler/../support/cmdline.h:64:74: error: no member named 'cmdline_error' in namespace 'cmdline'
  inline void throw_cmdline_error(const std::string &s) { throw cmdline::cmdline_error(s); }
                                                                ~~~~~~~~~^
1 error generated.
```

**Comment by xkszltl** [comment]
@steven-johnson 
I think you need to define the `cmdline_error` before this line, not after.

**Comment by steven-johnson** [comment]
Fixed via #4266
